app/javascript/components/Waybill.tsx

```diff
@@ -25,7 +26,35 @@ const Waybill = ({ currentUserRole }) => {
       componentMounted.current = false;
     };
   }, []);
-
+  const update_checkpoint_status = (id) => {
+    httpClient.route.get_routes(id).then((response) => setCheckpoints(response.data));
+    // let  new_checkpoints=checkpoints
+    // new_checkpoints.find(checkpoint => checkpoint.id == id).is_passed=value
+    // setCheckpoints(new_checkpoints)
+  };
+  const handleSubmit_waybill = (id) => {
+    httpClient.waybill.finish({ ids: id })
+      .then(() => {
+        const new_waybylls = waybills;
+        new_waybylls.find((waybill) => waybill.id == id).status = 'Delivered to the recipient';
```

Comment:
I guess its bad practice to change waybill status on front-end side in order to auto-refresh data. For example if back-end throws an error during update, waybill wouldnt change its status in DB, so on front-end status would change, but after refreshing page it will rollbacked to previous status. As an alternative i would suggest to send back in response to front-end request updated version of waybill object, and use setWaybill() with recieved data.